nbd/src/main.cpp

```diff
@@ -95,6 +97,112 @@ static void Usage() {
         << std::endl;
 }
 
+// use for record image and device info to auto map when boot
+static void AddRecord() {
+    std::fstream fs;
+    int fd;
+    fs.open(CURVETAB_PATH, std::fstream::in);
+    if (!fs.is_open()) {
+        std::cerr << "curve-nbd: open curvetab file failed.";
+        return;
+    }
+    fd = static_cast<__gnu_cxx::stdio_filebuf< char > * const >
+                                            (fs.rdbuf())->fd();
+    if (0 == flock(fd, LOCK_SH )) {
```

Comment:
If you want use the `flock` to sync the operation in file, i suggest use the `FileLock` class in [nebd/src/common/file_lock.cpp](https://github.com/opencurve/curve/blob/master/nebd/src/common/file_lock.cpp).